src/docfx/Errors.cs

```diff
@@ -469,10 +469,9 @@ public static Error MonikerRangeInvalid(SourceInfo? source, string message)
         /// which used monikerRange in its yaml header or used moniker-zone syntax.
         /// </summary>
         /// Behavior: ✔️ Message: ❌
-        public static Error MonikerRangeUndefined(SourceInfo? source, string? monikerRange)
+        public static Error MonikerRangeUndefined(SourceInfo<string?> source, FilePath file)
         {
-            var processedMonikerRange = monikerRange != null ? $"'{monikerRange}' " : string.Empty;
-            return new(ErrorLevel.Suggestion, "moniker-range-undefined", $"Moniker range {processedMonikerRange}is missing from the 'groups' setting in docfx.yml/docfx.json. It should not be defined in file metadata or moniker zone.", source);
+            return new(ErrorLevel.Suggestion, "moniker-range-undefined", $"Moniker range '{source}' should not be defined in file metadata of docfx.yml/docfx.json or in moniker zone of file '{file}'. Please check the 'groups' setting in docfx.yml/docfx.json.", source);
```

Comment:
The source already contains the file info, do we still need to mention it in the message?
If we still need, the source contains the FilePath property we don't need to pass it in? are you considering filemetadata or globalmetadata case?

Reply:
Yes, file metadata and global metadata are considered - you can find corresponding yaml test. Thus, here `source` could be `docfx.json`. However, since we'd like to print out the md file where this error happens, we will need to pass in `file` separately.